internal/server/data/selectors.go

```diff
@@ -118,6 +119,22 @@ func ByUserID(userID uid.ID) SelectorFunc {
 	}
 }
 
+func ByPagination(pg *models.Pagination) SelectorFunc {
+	if pg == nil {
+		defaultpg := models.RequestToPagination(nil)
+		pg = &defaultpg
+	}
+
+	return func(db *gorm.DB) *gorm.DB {
+		// var count int64
+		// db.Count(&count)
+
+		pg.MaxPage = -999999 //TODO: get maxpage working
+		// I don't think this is the right place to do this necessarily
+		return db.Offset(pg.Limit * (pg.Page - 1)).Limit(pg.Limit).Order(pg.Sort)
```

Comment:
This seems ok here to me, other place could be in the `data.List` function as well. 

What does `(pg.Page - 1)` do? Could this result in unexpected results if the request specifies a negative page number?

Reply:
No, because any number ≤ 0 gets set to page 1. Would it make more sense to send an error if someone passes a negative value or a value that's too large?

Reply:
I think an error makes sense 👍